python/raydp/spark/ray_cluster.py

```diff
@@ -67,11 +67,12 @@ def get_spark_session(self,
         except KeyError:
             extra_jars = []
         extra_conf["spark.jars"] = ",".join(glob.glob(RAYDP_CP) + extra_jars)
-        driver_cp = "spark.driver.extraClassPath"
-        if driver_cp in extra_conf:
-            extra_conf[driver_cp] = ":".join(glob.glob(RAYDP_CP)) + ":" + extra_conf[driver_cp]
+        driver_cp_key = "spark.driver.extraClassPath"
+        driver_cp = ":".join(glob.glob(RAYDP_CP) + glob.glob(RAY_CP))
+        if driver_cp_key in extra_conf:
+            extra_conf[driver_cp_key] = driver_cp + ":" + extra_conf[driver_cp_key]
         else:
-            extra_conf[driver_cp] = ":".join(glob.glob(RAYDP_CP))
+            extra_conf[driver_cp_key] = driver_cp
```

Comment:
We need to add this RAY_CP to spark's driver extra classpath because serialized owner_address is needed to register ownership in java, but we need to parse it to extract ip_address to use in spark. This might not be needed once we use cross language call to start our RayAppMaster, in that case we don't need to register ownership(?)